util/build_configs/cray-internal/generate-modulefile.bash

```diff
@@ -112,29 +145,45 @@ if { ! [ info exists env(PE_ENV) ] } {
     module load PrgEnv-gnu
 }
 
-##
-## Now, we're going to attempt to get to the right build type
-###
-#
-set network seastar
-if { [ info exists env(XTPE_NETWORK_TARGET) ] } {
-    set network $env(XTPE_NETWORK_TARGET)
-} elseif { [ info exists env(CRAYPE_NETWORK_TARGET) ] } {
-    set network $env(CRAYPE_NETWORK_TARGET)
-}
-set accel 0
-if { [ info exists env(CRAY_ACCEL_TARGET) ] } {
-    set accel 1
-}
+set compiler $env(PE_ENV)
 
+if { [string match cray-shasta $CHPL_HOST_PLATFORM] } {
+    # Interim settings for Shasta systems.
 
-setenv MPICH_GNI_DYNAMIC_CONN disabled
-if { [string match gemini $network] } {
-    set CHPL_HOST_PLATFORM cray-xe
-} elseif { [string match aries $network] } {
-    set CHPL_HOST_PLATFORM cray-xc
-} else {
-    set CHPL_HOST_PLATFORM cray-xt
+    # So far we only have gnu-based Chapel for Shasta.
```

Comment:
It seems wrong to me to swap the users PrgEnv- out from under them. I would prefer an error message instead

Reply:
It may have been overly subtle, but the "So far ..." was meant to indicate that this is only interim logic.  Eventually we'll presumably build the module with every PrgEnv, and then this else-clause will be entirely removed.